File: app/src/main/java/org/metabrainz/mobile/data/sources/api/CollectionService.java
package org.metabrainz.mobile.data.sources.api;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface CollectionService {

    @GET("collection")
    Call<ResponseBody> getPublicUserCollections(@Query("editor") String name);

    @GET("collection")
    Call<ResponseBody> getAllUserCollections(@Query("editor") String name,
                                             @Query("inc") String params);

    @GET("{entity}")
    Call<ResponseBody> getCollectionContents(@Path("entity") String entity,
                                             @Query("collection") String id);

}

File: app/src/main/java/org/metabrainz/mobile/data/sources/api/CollectionService.kt
package org.metabrainz.mobile.data.sources.api

import okhttp3.ResponseBody
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface CollectionService {
    @GET("collection")
    suspend fun getPublicUserCollections(@Query("editor") name: String): ResponseBody

    @GET("collection")
    suspend fun getAllUserCollections(@Query("editor") name: String,
                                      @Query("inc") params: String): ResponseBody

    @GET("{entity}")
    suspend fun getCollectionContents(@Path("entity") entity: String,
                                      @Query("collection") id: String): ResponseBody
}